### PR TITLE
add `check_repr_commutation_relation` to test the CSS orthogonality condition for 2BGA's Group Algebra with a General Group `G`

### PR DESCRIPTION
**Pull request by Fe-r-oz**
This PR implements the CSS orthogonality condition given by left and right representation matrix commutation relation of group algebra of the 2BGA code. This consistency check will be used in the `tests` PRs. This is inspired from insights from @thofma! 

Given that we are use a rich set of groups in the `tests` PRs , this consistency test will be provide an additional crosscheck.

Edit: 

Some details from literature

- In a group algebra $\mathbb{F}[G]$, the left $L(a)$ and right $R(b)$ representations of elements commute:  
  
  $$L(a) \cdot R(b) = R(b) \cdot L(a)$$

  This results from the associativity of multiplication.

The snapshot below talks about this in the following general way:  [Reference](https://arxiv.org/pdf/2111.03654)
- In an algebra $R$, the left $\lambda_r$ and right $\rho_r$ regular matrix representations commute:  
 
  $$\rho_a \lambda_b = \lambda_b \rho_a$$
 
  This follows from the associativity of multiplication in $R$.

Screenshot_select-area_20241024101141

### CODE REVIEW

**Comment by Fe-r-oz** [comment]
@Krastanov, This PR is ready for review. Thanks!  `check_repr_commutation_relation` will be used by the `tests` PRs to check the consistency of Group Algebra for 2BGA codes.

File: src/ecc/ECC.jl [hunk 1]
```diff
@@ -15,7 +15,7 @@ abstract type AbstractECC end
 
 export parity_checks, parity_checks_x, parity_checks_z, iscss,
     code_n, code_s, code_k, rate, distance,
-    isdegenerate, faults_matrix,
+    isdegenerate, faults_matrix, check_repr_commutation_relation,
```
Comment (Krastanov) [inline src/ecc/ECC.jl:18]: let's not export it for now, just keep it as an internal check

File: test/test_ecc_base.jl [hunk 2]
```diff
@@ -46,6 +46,7 @@ other_lifted_product_codes = []
 # [[882, 24, d≤24]] code from (B1) in Appendix B of [panteleev2021degenerate](@cite)
 l = 63
 GA = group_algebra(GF(2), abelian_group(l))
+@test check_repr_commutation_relation(GA) == true
```
Comment (Krastanov) [inline test/test_ecc_base.jl:49]: Is this check expected to be used anywhere else if merged?

Also, you do not need to do `== true` -- the result is already a boolean

**Review by Krastanov — CHANGES_REQUESTED** [review]
This looks useful, thanks! I left a few questions below

**Comment by Fe-r-oz** [comment]
I have incorporated your suggestions, Thanks!

File: ext/QuantumCliffordHeckeExt/util.jl [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+It verifies the commutation relation that states, `L(a)·R(b) = R(b)·L(a)`. This
+property shows that matrices from the left and right representation sets commute
+with each other, which is an important property related to the CSS orthogonality
+condition.
+"""
+function check_repr_commutation_relation(GA::GroupAlgebra)
+    a, b = rand(GA), rand(GA)
+    # Check commutation relation: L(a)R(b) = R(b)L(a)
+    L_a = representation_matrix(a)
+    R_a = representation_matrix(b, :right)
```
Comment (Fe-r-oz) [inline ext/QuantumCliffordHeckeExt/util.jl]: typo, It should be `R_b` not, `R_a`

File: src/ecc/codes/lifted_product.jl [hunk 4]
```diff
@@ -17,3 +17,6 @@ function generalized_bicycle_codes end
 
 """Implemented in a package extension with Hecke."""
 function bicycle_codes end
+
+"""Implemented in a package extension with Hecke."""
+function check_repr_commutation_relation end
```
Comment (Fe-r-oz) [inline src/ecc/codes/lifted_product.jl:22]: Move to `ECC\codes\util.jl` in ECC module

**Comment by Krastanov** [comment]
thanks, this will be of great use throughout the tests

**Pull request merged**